src/odemis/gui/comp/overlay/ellipse.py

```diff
@@ -76,6 +79,29 @@ def move_to(self, pos):
         self._phys_to_view()
         self.points.value = self._points
 
+    def get_state(self):
+        """Get the current state of the shape."""
+        state = {
+            EllipseStateKeys.P_POINT1: self.p_point1,
+            EllipseStateKeys.P_POINT2: self.p_point2,
+            EllipseStateKeys.P_POINT3: self.p_point3,
+            EllipseStateKeys.P_POINT4: self.p_point4,
+            EllipseStateKeys.POINTS_PRIVATE: self._points.copy(),
+        }
+        # Only return the state if the ellipse creation is finished
+        # By doing so avoid storing an undo action during ellipse creation
+        return state if self.p_point1 != self.p_point3 else {}
+
+    def restore_state(self, state: Dict[EllipseStateKeys, Any]):
+        """Restore the shape to a given state."""
+        self.p_point1 = state[EllipseStateKeys.P_POINT1]
+        self.p_point2 = state[EllipseStateKeys.P_POINT2]
+        self.p_point3 = state[EllipseStateKeys.P_POINT3]
+        self.p_point4 = state[EllipseStateKeys.P_POINT4]
+        self._points = state[EllipseStateKeys.POINTS_PRIVATE]
```

Comment:
The first thing that happens in `draw()` is `self._points.clear()`. They are dependent only on the 4 p_points. So what's the point of storing them? Is it because storing them is easier than recomputing them? If so, please add a comment explaining this. Otherwise, don't store them, and force the recomputation.

Reply:
I store them, just to set the `points` VA in line 103. Settings the points VA is important because it does the megafield computation in the subscribers callback. If I don't store them, I have to call `update_drawing()` first and then I can set the `points` VA. Doing this will call `update_drawing` twice, once here and then in ShapesOverlay's `undo` or `redo`